examples/campaign_management/add_complete_campaigns_using_batch_job.rb

```diff
@@ -0,0 +1,370 @@
+#!/usr/bin/env ruby
+# Encoding: utf-8
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This example adds complete campaigns including campaign budgets, campaigns,
+# ad groups and keywords using BatchJobService.
+
+require 'optparse'
+require 'google/ads/google_ads'
+
+def add_complete_campaigns_using_batch_job(customer_id)
+  # GoogleAdsClient will read a config file from
+  # ENV['HOME']/google_ads_config.rb when called without parameters
+  client = Google::Ads::GoogleAds::GoogleAdsClient.new
+
+  batch_job_service_client = client.service.batch_job
+
+  batch_job_resource_name = create_batch_job(
+    client,
+    batch_job_service_client,
+    customer_id,
+  )
+
+  add_all_batch_job_operations(
+    client,
+    batch_job_service_client,
+    customer_id,
+    batch_job_resource_name,
+  )
+
+  operation_response = run_batch_job(
+    batch_job_service_client,
+    batch_job_resource_name,
+  )
+
+  poll_batch_job(operation_response)
+
+  fetch_and_print_results(batch_job_service_client, batch_job_resource_name)
+end
+
+# Creates a new batch job for the specified customer ID.
+def create_batch_job(client, batch_job_service_client, customer_id)
+  # Creates a batch job operation to create a new batch job.
+  operation = client.operation.create_resource.batch_job do |bj|
+  end
```

Comment:
I don't think you need to use a block at all if you're not doing anything with it. Just client.operation.create_resource.batch_job should be sufficient.

Reply:
The operation would be empty if I don't use a block here. Do you want me to file a bug?
(I've now added a comment here to explain the empty block)

Reply:
Yeah, you shouldn't need to pass an empty block, so that seems like a bug to me.

Reply:
Created https://github.com/googleads/google-ads-ruby/issues/227 to track it